_scripts/app1-8.R

```diff
@@ -0,0 +1,84 @@
+# Part A ----
+
+{
+  
+  rm(list=ls());
+  options(show.error.locations = TRUE);
+  
+  # Prompt user to input temperature
+  temp <- as.numeric(readline("What is the temperature? "))
+  
+  # check if the users value is greater than -20 but less than 30
+  if(temp > -20 & temp < 30)
+  {
+    cat("cold");
+  }
+  # check if the users value is equal to 30
+  else if(temp == 30)
+  {
+    cat("cold-ish");
+  }
+  # check if the users value is equal to 60
+  else if(temp == 60)
+  {
+    cat("hot-ish");
+  }
+  # check if the users value is greater than 60 but less than 100
+  else if(temp > 60 & temp <= 100)
+  {
+    cat("hot");
+  }
+  # check if the user has entered values outside the ranges listed above:
+  # E.g. < -20 & > 100
+  else 
+  {
+    cat("invalid value")
+  }
+
```

Comment:
You cheated and used the logical operator & that is taught in the next lesson.  Which is fine -- in this case you have conditions that can be combined (like temp > 60 & temp <= 100) because they lead to the same outcome.  

You use & whereas I use &&.  For single values, the & ans && operators function exactly the same.  For vectors, the & operates on values individually whereas && operates on the whole vector.  This is covered in the next unit.

The only reason I prefer to use && for single values is that this is more in line with other programming languages.

Reply:
Oops! I actually wasn't aware that I was getting ahead ha. During the lesson I was doing some additional reading on carrying out Boolean operations in R.

Reply:
I still think it is good practice to do part A without using the combinatory logic (i.e., the &).  There are situations where every condition requires a different outcome and can be a little tricky to do efficiently when they involve continuous numbers (emphasis on efficiently).